src/lightmap.cpp

```diff
@@ -137,7 +141,13 @@ bool map::build_transparency_cache( const int zlev )
                         zero_value = value;
                         continue;
                     }
-                    for( const auto &fld : cur_submap->get_field( { sx, sy } ) ) {
+
+                    field f = cur_submap->get_field( { sx, sy } );
+                    if( !f.displayed_field_type() ) {
```

Comment:
I'm really surprised that this check is was faster than iterating over an empty container.  Did you profile this change in isolation?  I suppose it might be improving the branch prediction...

Reply:
Good idea! I'll do that tomorrow.

Reply:
After reading #44042 and #44013 I was wondering if perhaps not all fields caused `displayed_field_type` to be non-null.  If there were fields that are not displayed, and this allows us to skip iterating over those fields, then this would make sense as an optimization.  But reading the code of `field::add_field` it looks like `_displayed_field_type` is always updated for any added field; it doesn't appear to check whether that field should be displayed.

Reply:
You are right, before #44013 `_displayed_field_type` was set to non-null value only if added field's `priority` was > 0. Note, that it if field priority is < 0 it doesn't mean that this field should not be processed.

In #44013 I changed `_displayed_field_type`  semantics to contain the field with maximal priority even if it's less than zero. Currently, if `_displayed_field_type == fd_null` it means that field list is empty. I think you're right and checking the `_displayed_field_type` in comparable in performance to `field.begin()` and performance improvement comes from the fact that majority of fields are empty, so it's `!_displayed_field_type` vs `field.begin()` + `field.end()`.

I'll so some more research tomorrow. Maybe there is a way to make things less confusing and reasonably fast at the same time.

Reply:
While we're micro-optimizing: `field.begin()` and `field.end()` are not currently `inline`, but could easily be.